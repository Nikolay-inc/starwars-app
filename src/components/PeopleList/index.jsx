// to have a chance to use Chakra UI
'use client';

import ListItem from '../ListItem';
import { ChakraProvider, Accordion } from '@chakra-ui/react';

const PeopleList = ({ data }) => {
  return (
    <div>
      <ChakraProvider>
        {/* map people to make a list items */}
        {data.map((person) => (
          <Accordion allowMultiple key={person.name}>
            <ListItem person={person} />
          </Accordion>
        ))}
      </ChakraProvider>
    </div>
  );
  }
  
  export default PeopleList;