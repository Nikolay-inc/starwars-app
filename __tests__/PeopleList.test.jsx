import { render, screen } from '@testing-library/react';
import PeopleList from '../src/components/PeopleList';

const mockData = [
  {
    name: 'Luke Skywalker',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 1,
    films: [1, 2, 3, 6],
    species: [1],
    vehicles: [14, 30],
    starships: [12, 22],
    created: '2014-12-09T13:50:51.644000Z',
    edited: '2014-12-20T21:17:56.891000Z',
    url: 'https://sw-api.starnavi.io/people/1/'
  },
];

describe('PeopleList component', () => {
  test('renders list of people', () => {
    render(<PeopleList data={mockData} />);
    
    // ensure each person's name is rendered
    mockData.forEach(person => {
      const personName = screen.getByText(person.name);
      expect(personName).toBeInTheDocument();
    });
  });

  test('renders person details within each accordion', () => {
    render(<PeopleList data={mockData} />);
    
    mockData.forEach(person => {
      expect(screen.getByText(`Height - ${person.height}`)).toBeInTheDocument();
      expect(screen.getByText(`Mass - ${person.mass}`)).toBeInTheDocument();
    });
  });

  test('renders nothing when data is empty', () => {
    render(<PeopleList data={[]} />);
    
    // ensure there are no accordions rendered
    expect(screen.queryByRole('tabpanel')).not.toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(
      <PeopleList data={mockData} />
    );
    expect(container).toMatchSnapshot();
  });
});
