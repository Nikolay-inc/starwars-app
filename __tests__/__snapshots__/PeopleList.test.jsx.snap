// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleList component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="chakra-accordion css-0"
    >
      <div
        class="chakra-accordion__item css-5kkxfh"
      >
        <h2>
          <button
            aria-controls="accordion-panel-:r4:"
            aria-expanded="false"
            class="chakra-accordion__button css-uttm9k"
            data-index="0"
            id="accordion-button-:r4:"
            type="button"
          >
            <span
              class="css-1eziwv"
            >
              <a
                class="chakra-link css-g9rydc"
                href="/people/1"
              >
                Luke Skywalker
              </a>
            </span>
            <svg
              aria-hidden="true"
              class="chakra-icon chakra-accordion__icon css-186l2rg"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r4:"
            class="chakra-accordion__panel css-1hb5ode"
            id="accordion-panel-:r4:"
            role="region"
          >
            <div>
              <div
                class="chakra-stack css-j7qwjs"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-wt5l11"
                  >
                    Height - 
                    172
                  </p>
                  <p
                    class="chakra-text css-wt5l11"
                  >
                    Mass - 
                    77
                  </p>
                </div>
                <div
                  class="chakra-stack__divider css-1q5cbl"
                />
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-wt5l11"
                  >
                    Hair Color - 
                    blond
                  </p>
                  <p
                    class="chakra-text css-wt5l11"
                  >
                    Skin Color - 
                    fair
                  </p>
                  <p
                    class="chakra-text css-wt5l11"
                  >
                    Eye Color - 
                    blue
                  </p>
                </div>
                <div
                  class="chakra-stack__divider css-1q5cbl"
                />
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-wt5l11"
                  >
                    Birth Year - 
                    19BBY
                  </p>
                  <p
                    class="chakra-text css-wt5l11"
                  >
                    Gender - 
                    male
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>
</div>
`;
